lib/codecs/src/encoding/format/syslog.rs

```diff
@@ -0,0 +1,446 @@
+            if let Some(field_name) = value.strip_prefix("$.message.") {
+                return Ok(Facility::Field(field_name.to_string()));
+            } else {
+                let num = match value.to_uppercase().as_str() {
+                    "KERN" => 0,
+                    "USER" => 1,
+                    "MAIL" => 2,
+                    "DAEMON" => 3,
+                    "AUTH" => 4,
+                    "SYSLOG" => 5,
+                    "LPR" => 6,
+                    "NEWS" => 7,
+                    "UUCP" => 8,
+                    "CRON" => 9,
+                    "AUTHPRIV" => 10,
+                    "FTP" => 11,
+                    "NTP" => 12,
+                    "SECURITY" => 13,
+                    "CONSOLE" => 14,
+                    "SOLARIS-CRON" => 15,
+                    "LOCAL0" => 16,
+                    "LOCAL1" => 17,
+                    "LOCAL2" => 18,
+                    "LOCAL3" => 19,
+                    "LOCAL4" => 20,
+                    "LOCAL5" => 21,
+                    "LOCAL6" => 22,
+                    "LOCAL7" => 23,
+                    _ => 24,
+                };
+                if num > 23 {
+                    return Err(de::Error::invalid_value(de::Unexpected::Unsigned(num as u64), &"unknown facility"));
+                } else {
+                    return Ok(Facility::Fixed(num))
+                }
+            }
+        }
+    }
+}
+
+fn deserialize_severity<'de, D>(d: D) -> Result<Severity, D::Error>
+    where D: de::Deserializer<'de>
+{
+    let value: String = String::deserialize(d)?;
+    let num_value = value.parse::<u8>();
+    match num_value {
+        Ok(num) => {
+            if num > 7 {
+                return Err(de::Error::invalid_value(de::Unexpected::Unsigned(num as u64), &"severity number too large"))
+            } else {
+                return Ok(Severity::Fixed(num))
+            }
+        }
+        Err(_) => {
+            if let Some(field_name) = value.strip_prefix("$.message.") {
+                return Ok(Severity::Field(field_name.to_string()));
+            } else {
+                let num = match value.to_uppercase().as_str() {
+                    "EMERGENCY" => 0,
+                    "ALERT" => 1,
+                    "CRITICAL" => 2,
+                    "ERROR" => 3,
+                    "WARNING" => 4,
+                    "NOTICE" => 5,
+                    "INFORMATIONAL" => 6,
+                    "DEBUG" => 7,
+                    _ => 8,
+                };
+                if num > 7 {
+                    return Err(de::Error::invalid_value(de::Unexpected::Unsigned(num as u64), &"unknown severity"))
+                } else {
+                    return Ok(Severity::Fixed(num))
+                }
+            }
+        }
+    }
+}
+
+fn default_app_name() -> String {
+    String::from("vector")
+}
+
+fn default_nil_value() -> String {
+    String::from(NIL_VALUE)
+}
+
+fn add_log_source(log: &LogEvent, buf: &mut String) {
```

Comment:
I'm not sure quite what this is trying to do, but I'm pretty sure it should be removed. If users want these details as part of the message they can use `vrl` to construct it.

Reply:
@StephenWakely What do you mean by "field with a meaning of X"?

Reply:
So, fields can be given a meaning. This allows us to have fields coming from different sources with different names, but we are still able to get the field by their meaning without worrying about what their name is.

For example, the syslog source creates a field called [`hostname`](https://github.com/vectordotdev/vector/blob/master/lib/codecs/src/decoding/format/syslog.rs#L161-L167), the journald source creates a field called [`host`](https://github.com/vectordotdev/vector/blob/master/src/sources/journald.rs#L285-L291). Both mean the same thing, but have different names.

The Papertrail sink needs to use the host field in its message. So rather than get the field by it's name it get's it by the meaning. See [here](https://github.com/vectordotdev/vector/blob/master/src/sinks/papertrail.rs#L141-L144). 

This can also be managed in VRL with the [`set_semantic_meaning` function](https://vector.dev/docs/reference/vrl/functions/#set_semantic_meaning).

You'll notice in [get_host](https://github.com/vectordotdev/vector/blob/master/lib/vector-core/src/event/log_event.rs#L580) that each log message can be `Vector` or `Legacy`. The concept of meaning is in transition and only applies to the Vector namespace. There is an RFC that goes into it [here](https://github.com/vectordotdev/vector/blob/master/rfcs/2022-04-20-12187-log-namespacing.md). A long term goal for us is to eventually remove the legacy namespace - but that is probably a long way away. 

For the purposes of this PR, I would call the functions in `log_event` where there exists one - eg [`get_message`](https://github.com/vectordotdev/vector/blob/master/lib/vector-core/src/event/log_event.rs#L550). For other fields, eg `procid`, we can just hardcode the field names for the Legacy namespace. 

eg.
```
        match self.namespace() {
            LogNamespace::Vector => self.get_by_meaning("procid"),
            LogNamespace::Legacy => self.get("procid"),
        }
```

Base it on the names set by the [syslog decoder](https://github.com/vectordotdev/vector/blob/master/lib/codecs/src/decoding/format/syslog.rs#L213).

Hope that makes sense. Any problems, give me a shout..